In [ ]:
#!pip install plotly_express
#!pip install pandas 

In [ ]:
# import libraries
import pandas as pd 
import plotly_express as px 

In [ ]:
# first dataset
try:
    url1 = 'datasets/tinder.csv'
    tinder = pd.read_csv(url1, on_bad_lines='skip')
    print('local')
except:
    print('remote')
    url1 = 'https://raw.githubusercontent.com/dataPracticum/code_pudding_v2/main/Tinder.csv'
    tinder = pd.read_csv(url1, on_bad_lines='skip', names=['id', 'usage', 'university', 'answer', 'matches', 'percentage', 'relationship'])

local


In [ ]:
tinder.head()

id   usage       university answer     matches  percentage  \
0  292890.8970     web  Meridian, Idaho     No    0.000000    0.000000   
1  292887.9870     web  Meridian, Idaho     No    0.000000    0.000000   
2  292894.0656  gender  Meridian, Idaho     No  499.173606    0.225255   
3  292887.1180     web  Meridian, Idaho     No    0.000000    0.000000   
4  292893.6561  gender  Meridian, Idaho     No  455.925963    0.211360   

   relationship  
0             0  
1             0  
2             0  
3             0  
4             0

In [ ]:
# function to convert to json
def convert_json(df):
    df_json = df.to_json()
    data = json.loads(df_json)
    new_df = json.dumps(data, indent=2)
    return print(new_df)

# Introduction

## Tinder

In [ ]:
# look at data
tinder.head()

id   usage       university answer     matches  percentage  \
0  292890.8970     web  Meridian, Idaho     No    0.000000    0.000000   
1  292887.9870     web  Meridian, Idaho     No    0.000000    0.000000   
2  292894.0656  gender  Meridian, Idaho     No  499.173606    0.225255   
3  292887.1180     web  Meridian, Idaho     No    0.000000    0.000000   
4  292893.6561  gender  Meridian, Idaho     No  455.925963    0.211360   

   relationship  
0             0  
1             0  
2             0  
3             0  
4             0

In [ ]:
# missing values
tinder.isna().sum()

id              0
usage           0
university      0
answer          0
matches         0
percentage      0
relationship    0
dtype: int64

In [ ]:
# looking for duplicates
tinder[tinder.duplicated()]

id usage       university answer  matches  percentage  \
1295  292889.9062   web  Meridian, Idaho     No      0.0         0.0   
1636  292887.7623   web  Meridian, Idaho     No      0.0         0.0   

      relationship  
1295             0  
1636             0

Different id's so they are not actually duplicates

In [ ]:
# values of answer
tinder.answer.value_counts()

No                    1571
Yes                    164
I don't use Tinder     161
Name: answer, dtype: int64

In [ ]:
# tinder columns
tinder.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1896 entries, 0 to 1895
Data columns (total 7 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   id            1896 non-null   float64
 1   usage         1896 non-null   object 
 2   university    1896 non-null   object 
 3   answer        1896 non-null   object 
 4   matches       1896 non-null   float64
 5   percentage    1896 non-null   float64
 6   relationship  1896 non-null   int64  
dtypes: float64(3), int64(1), object(3)
memory usage: 103.8+ KB


Tinder is a relatively small dataset with 1,896 values. Target value is relationship, where it determines whether the individual found a relationship using the app. We can use this data to run a ML model to predict wether an individual will get a relationship from Tinder. 

### EDA

In [ ]:
# correlation
tinder.corr()

id   matches  percentage  relationship
id            1.000000 -0.429790    0.164302      0.167766
matches      -0.429790  1.000000    0.412200      0.186139
percentage    0.164302  0.412200    1.000000      0.311429
relationship  0.167766  0.186139    0.311429      1.000000

In [ ]:
# summary statistics
tinder.describe()

id      matches   percentage  relationship
count    1896.000000  1896.000000  1896.000000   1896.000000
mean   292999.394750   446.881267     0.254098      0.350738
std       227.809806   494.110578     0.220809      0.477327
min    292881.000000     0.000000     0.000000      0.000000
25%    292888.196925     0.000000     0.000000      0.000000
50%    292893.904400   400.439633     0.257873      0.000000
75%    292902.568750   845.560798     0.396787      1.000000
max    293785.000000  1969.000000     1.000000      1.000000

In [ ]:
# Skew
tinder.skew()

C:\Users\XIX\AppData\Local\Temp\ipykernel_25460\3200751176.py:2: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.
  tinder.skew()


id              2.044382
matches         0.716367
percentage      0.816940
relationship    0.626066
dtype: float64

In [ ]:
# answers to question: do you use tinder?
tinder.answer.value_counts()

No                    1571
Yes                    164
I don't use Tinder     161
Name: answer, dtype: int64

In [ ]:
# compile no answers
tinder.answer.replace('I don\'t use Tinder', 'No', inplace=True)

In [ ]:
# answers to question: do you use tinder?
tinder.answer.value_counts()

No     1732
Yes     164
Name: answer, dtype: int64

In [ ]:
# use tinder and found relationship
tinder[tinder.answer=='Yes'].relationship.value_counts()

1    86
0    78
Name: relationship, dtype: int64

In [ ]:
# relationship found using tinder 
print(86/(86+78) * 100, '%')

52.4390243902439 %


In [ ]:
# dont use tinder
tinder[tinder.answer=='No'].relationship.value_counts()

0    1153
1     579
Name: relationship, dtype: int64

In [ ]:
# relationship found not using tinder
print(579/(1153+579) * 100, '%')

33.429561200923786 %


In [ ]:
# dataframe comparing relationship success
data = {'category': ['Tinder', 'Not Tinder'], 'percentage':[52.44, 33.43]}
comparison = pd.DataFrame(data)

In [ ]:
px.bar(comparison, x='category', y='percentage', color='category', title='Relationship Success Rates')

In [ ]:
# university names
tinder.university.value_counts()

Meridian, Idaho                       1421
Westport, CT                            14
University of Pennsylvania              12
University of Mississippi               10
University of Washington                10
                                      ... 
Boston College                           1
Southern Methodist University            1
Mississippi State University             1
Washington University in St. Louis       1
North Carolina State                     1
Name: university, Length: 148, dtype: int64

In [ ]:
# medium of usage
tinder.usage.value_counts()

gender        488
mobile        483
university    464
web           461
Name: usage, dtype: int64

In [ ]:
# target values
tinder.relationship.value_counts()

0    1231
1     665
Name: relationship, dtype: int64

In [ ]:
px.pie(tinder.answer, names='answer', title='Total Percentage of Tinder Users', template='plotly_white', height=600)

More people do not use Tinder

In [ ]:
px.pie(tinder.relationship, names='relationship', title='Total Percentage of Relationships', template='plotly_white', height=600)

More people are not ina relationship

In [ ]:
px.pie(tinder[tinder.answer=='Yes'].relationship, names='relationship', title='Relationships Found Using Tinder', template='plotly_white', height=600)

A little more than half of the people that use tinder have found a relationship using the app

In [ ]:
px.pie(tinder[tinder.answer=='No'].relationship, names='relationship', title='Found a Relationship Without Tinder', template='plotly_white', height=600,)


A third of People not using Tinder have found a relationship

In [ ]:
columns = ['usage', 'answer', 'matches', 'percentage',
       'relationship']
for column in columns:
    px.histogram(tinder[column], title='Distribution of '+ str.upper(column), labels={'value': str.upper(column)}, template='plotly_white', color_discrete_sequence=['red']).show()

In [ ]:
# number of unique universities
tinder.university.nunique()

148

In [ ]:
tinder.university.value_counts().nlargest(20)

Meridian, Idaho                         1421
Westport, CT                              14
University of Pennsylvania                12
University of Mississippi                 10
University of Washington                  10
University of Southern California          9
Washington State University                8
University of Tampa                        8
Wesleyan University                        8
University of Wisconsin, Eau Claire        8
University of California, Santa Cruz       8
University of California, Berkeley         8
Williams College                           7
Vassar College                             7
Vanderbilt University                      7
University of Buffalo                      7
University of Oregon                       7
Union College                              7
Whatsgoodly University                     6
University of Delaware                     6
Name: university, dtype: int64

In [ ]:
px.bar(tinder.university.value_counts().nlargest(20), title='Distribution of Most Common Universities', labels={'value': 'Universities'}, template='plotly_white', color_discrete_sequence=['red']).show()

Most common university is Meridian in Ohio. The other universities make up a small percentage, roughly 15%  of the rest of the schools.

In [ ]:
columns = ['matches', 'percentage']
for column in columns:
    px.box(tinder[column], title='Distribution of '+ str.upper(column), labels={'value': str.upper(column)}, template='plotly_white', color_discrete_sequence=['red']).show()

Average matches are around 400, with the max at 1969. The minimm amount of matches is 0, while the upper quartile is 846. Most people get matches on .25% of swipes, while the interquartile range is 0-0.4%.  

In [ ]:
px.imshow(tinder.corr(), text_auto=True, title='Tinder Correlations', template='plotly_white', aspect='equal', height=600, color_continuous_scale='hot')

# Looking for Love in All the Wrong Places?

* Are you looking for love? What are your chances of finding it on tinder, and what are your chances of finding it by yourself? 
* Can we predict your chances on tinder based on your school and average stats?


## ML Predicting Relationship

In [ ]:
from sklearn.linear_model import LogisticRegression
from sklearn.ensemble import RandomForestClassifier
from sklearn.tree import DecisionTreeClassifier
from sklearn import svm
from sklearn.naive_bayes import GaussianNB
from sklearn.neighbors import KNeighborsClassifier
from sklearn.pipeline import Pipeline
from sklearn.model_selection import cross_val_score, train_test_split
from sklearn.preprocessing import StandardScaler, OrdinalEncoder, LabelEncoder

In [ ]:
encoder = OrdinalEncoder()
data_ordinal = pd.DataFrame(encoder.fit_transform(tinder), columns=tinder.columns)

X = data_ordinal.drop(columns=['relationship', 'id'])
y = data_ordinal.relationship



In [ ]:
X.head()

usage  university  answer  matches  percentage
0    3.0        46.0     0.0      0.0         0.0
1    3.0        46.0     0.0      0.0         0.0
2    0.0        46.0     0.0    275.0       157.0
3    3.0        46.0     0.0      0.0         0.0
4    0.0        46.0     0.0    201.0       100.0

In [ ]:
y.value_counts()

0.0    1231
1.0     665
Name: relationship, dtype: int64

#### Usage Map

In [ ]:
# show  map of coded values
tinder.usage.value_counts()

gender        488
mobile        483
university    464
web           461
Name: usage, dtype: int64

In [ ]:
# label encoded values
X.usage.value_counts()

0.0    488
1.0    483
2.0    464
3.0    461
Name: usage, dtype: int64

#### University Map

In [ ]:
# show  map of coded values
tinder.university.value_counts().nlargest(20)

Meridian, Idaho                         1421
Westport, CT                              14
University of Pennsylvania                12
University of Mississippi                 10
University of Washington                  10
University of Southern California          9
Washington State University                8
University of Tampa                        8
Wesleyan University                        8
University of Wisconsin, Eau Claire        8
University of California, Santa Cruz       8
University of California, Berkeley         8
Williams College                           7
Vassar College                             7
Vanderbilt University                      7
University of Buffalo                      7
University of Oregon                       7
Union College                              7
Whatsgoodly University                     6
University of Delaware                     6
Name: university, dtype: int64

In [ ]:
# coded values
X.university.value_counts().nlargest(20)

46.0     1421
144.0      14
123.0      12
117.0      10
133.0      10
127.0       9
140.0       8
129.0       8
143.0       8
134.0       8
97.0        8
92.0        8
146.0       7
139.0       7
138.0       7
91.0        7
122.0       7
86.0        7
145.0       6
104.0       6
Name: university, dtype: int64

May be easier to assume Meridian, as it is the most prevalent, or at least one of the top 20 universities.

In [ ]:
# new data to make predictions
new_data = pd.DataFrame(index=['usage', 'university', 'answer', 'matches', 'percentage'], data=[3, 46, 1, 400, 0.25])
new_data = new_data.T

In [ ]:
new_data

usage  university  answer  matches  percentage
0    3.0        46.0     1.0    400.0        0.25

In [ ]:
# Classifier pipeline
pipe_lr = Pipeline([('scalar1', StandardScaler()), ('lr_classifier', LogisticRegression(random_state=19, max_iter=1000))])
pipe_dt = Pipeline([('scalar1', StandardScaler()), ('dt_classifier', DecisionTreeClassifier(random_state=19))])
pipe_rf = Pipeline([('scalar1', StandardScaler()), ('rf_classifier', RandomForestClassifier(random_state=19))])
pipe_sv = Pipeline([('scalar1', StandardScaler()), ('svm_classifier', svm.SVC(random_state=19))])
pipe_nb = Pipeline([('scalar1', StandardScaler()), ('nb_classifier', GaussianNB())])
pipe_kn = Pipeline([('scalar1', StandardScaler()), ('knn_classifier', KNeighborsClassifier(n_neighbors=3))])

pipelines = [pipe_lr, pipe_dt, pipe_rf, pipe_sv, pipe_nb, pipe_kn]

best_accuracy = 0
best_classifier = 0
best_pipeline = ""

pipe_dict = {0: 'Logistic Regression', 1: 'Decision Tree', 2: 'Random Forest', 3: 'SVM', 4: 'Naive-Bayes', 5: 'KNN'}

# Use cross-validation to evaluate the models
for i, model in enumerate(pipelines):
    scores = cross_val_score(model, X, y, cv=5, scoring='f1')
    print('{} Cross-Validation Accuracy: {:.2f}'.format(pipe_dict[i], scores.mean()))
    if scores.mean() > best_accuracy:
        best_accuracy = scores.mean()
        best_pipeline = model
        best_classifier = i

# Print the best classifier
print('\nClassifier with the best accuracy: {}'.format(pipe_dict[best_classifier]))

Logistic Regression Cross-Validation Accuracy: 0.41
Decision Tree Cross-Validation Accuracy: 0.83
Random Forest Cross-Validation Accuracy: 0.85
SVM Cross-Validation Accuracy: 0.83
Naive-Bayes Cross-Validation Accuracy: 0.52
KNN Cross-Validation Accuracy: 0.85

Classifier with the best accuracy: Random Forest


In [ ]:
# XGB

import xgboost as xgb

X2 = tinder.drop(columns=['relationship', 'id'])
y2 = tinder.relationship

lbl = LabelEncoder()
X2.usage = lbl.fit_transform(X2.usage.astype(str))
X2.university = lbl.fit_transform(X2.university.astype(str))
X2.answer = lbl.fit_transform(X2.answer.astype(str))
X2.matches = lbl.fit_transform(X2.matches.astype(str))
X2.percentage = lbl.fit_transform(X2.percentage.astype(str))



X2_train, X2_test, y2_train, y2_test = train_test_split(
    X2, y2, test_size=0.25, random_state=19)

# Create a XGBM 
xgbc = xgb.XGBClassifier(random_state=19)

xgbc.fit(X2_train, y2_train)

scores = cross_val_score(xgbc, X2, y2, cv=5, scoring='f1') 
final_score = sum(scores) / len(scores)
print('Average model evaluation score:', final_score)

Average model evaluation score: 0.8586813125972839


In [ ]:
# new data to make predictions with categorical columns
new_data2 = pd.DataFrame(index=['usage', 'university', 'answer', 'matches', 'percentage'], data=['web','Meridian, Idaho' , 'Yes', 400, 0.25])
new_data2 = new_data2.T

In [ ]:
new_data2.matches = new_data2.matches.astype('float')
new_data2.percentage = new_data2.percentage.astype('float')
new_data2.answer = new_data2.answer.astype('category')
new_data2.university = new_data2.university.astype('category')
new_data2.usage = new_data2.usage.astype('category')

In [ ]:
new_data2.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1 entries, 0 to 0
Data columns (total 5 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   usage       1 non-null      object 
 1   university  1 non-null      object 
 2   answer      1 non-null      object 
 3   matches     1 non-null      float64
 4   percentage  1 non-null      float64
dtypes: float64(2), object(3)
memory usage: 168.0+ bytes


In [ ]:
xgbc.predict(new_data2)

ValueError: DataFrame.dtypes for data must be int, float, bool or category. When categorical type is supplied, The experimental DMatrix parameter`enable_categorical` must be set to `True`.  Invalid columns:usage: object, university: object, answer: object

In [ ]:
from sklearn.model_selection import GridSearchCV
from xgboost import XGBClassifier

param_grid = {
    'max_depth': [3, 4, 5],
    'learning_rate': [0.1, 0.2, 0.3],
    'n_estimators': [100, 200, 300],
}

# Create the XGBoost model
xgb = XGBClassifier()

# Create the GridSearchCV object
grid_search = GridSearchCV(xgb, param_grid, cv=5, scoring='f1', verbose=3, n_jobs=-1)


# Fit the GridSearchCV object to the data
grid_search.fit(X, y)

scores = cross_val_score(grid_search.best_estimator_, X, y, cv=5, scoring='f1') 
final_score = sum(scores) / len(scores)
print('Average model evaluation score:', final_score)


Fitting 5 folds for each of 27 candidates, totalling 135 fits
Average model evaluation score: 0.868553821658959


In [ ]:
# prediction of relationship status based on new data
grid_search.best_estimator_.predict(new_data)

array([0])

Prediction is single based on the new data

In [ ]:
# catboost

from catboost import CatBoostClassifier, Pool


catb = CatBoostClassifier(task_type='GPU', loss_function='Logloss', eval_metric='AUC', iterations=1500, random_seed=19)

catb.fit(X, y, verbose=100, use_best_model=True, plot=True)

scores = cross_val_score(catb, X, y, cv=2, scoring='f1') 
final_score = sum(scores) / len(scores)
print('Average model evaluation score:', final_score)

MetricVisualizer(layout=Layout(align_self='stretch', height='500px'))

You should provide test set for use best model. use_best_model parameter has been switched to false value.
Default metric period is 5 because AUC is/are not implemented for GPU


Learning rate set to 0.022632
0:	total: 61ms	remaining: 1m 31s
100:	total: 7.58s	remaining: 1m 45s
200:	total: 13.6s	remaining: 1m 27s
300:	total: 18.8s	remaining: 1m 14s
400:	total: 24.2s	remaining: 1m 6s
500:	total: 29.2s	remaining: 58.3s
600:	total: 34.1s	remaining: 51s
700:	total: 39.4s	remaining: 44.9s
800:	total: 44.3s	remaining: 38.6s
900:	total: 49.2s	remaining: 32.7s
1000:	total: 54.5s	remaining: 27.2s
1100:	total: 59.9s	remaining: 21.7s
1200:	total: 1m 4s	remaining: 16.1s
1300:	total: 1m 9s	remaining: 10.6s
1400:	total: 1m 14s	remaining: 5.24s
1499:	total: 1m 18s	remaining: 0us
Learning rate set to 0.022912


Default metric period is 5 because AUC is/are not implemented for GPU


0:	total: 45.3ms	remaining: 1m 7s
1:	total: 89.3ms	remaining: 1m 6s
2:	total: 132ms	remaining: 1m 5s
3:	total: 182ms	remaining: 1m 8s
4:	total: 227ms	remaining: 1m 7s
5:	total: 281ms	remaining: 1m 10s
6:	total: 333ms	remaining: 1m 11s
7:	total: 378ms	remaining: 1m 10s
8:	total: 428ms	remaining: 1m 10s
9:	total: 474ms	remaining: 1m 10s
10:	total: 522ms	remaining: 1m 10s
11:	total: 572ms	remaining: 1m 10s
12:	total: 623ms	remaining: 1m 11s
13:	total: 675ms	remaining: 1m 11s
14:	total: 722ms	remaining: 1m 11s
15:	total: 772ms	remaining: 1m 11s
16:	total: 824ms	remaining: 1m 11s
17:	total: 870ms	remaining: 1m 11s
18:	total: 926ms	remaining: 1m 12s
19:	total: 975ms	remaining: 1m 12s
20:	total: 1.02s	remaining: 1m 11s
21:	total: 1.07s	remaining: 1m 12s
22:	total: 1.12s	remaining: 1m 11s
23:	total: 1.2s	remaining: 1m 13s
24:	total: 1.25s	remaining: 1m 13s
25:	total: 1.3s	remaining: 1m 13s
26:	total: 1.34s	remaining: 1m 13s
27:	total: 1.4s	remaining: 1m 13s
28:	total: 1.44s	remaining: 1m 13s
2

Default metric period is 5 because AUC is/are not implemented for GPU


0:	total: 49.3ms	remaining: 1m 13s
1:	total: 99.3ms	remaining: 1m 14s
2:	total: 149ms	remaining: 1m 14s
3:	total: 196ms	remaining: 1m 13s
4:	total: 247ms	remaining: 1m 13s
5:	total: 298ms	remaining: 1m 14s
6:	total: 345ms	remaining: 1m 13s
7:	total: 385ms	remaining: 1m 11s
8:	total: 438ms	remaining: 1m 12s
9:	total: 489ms	remaining: 1m 12s
10:	total: 545ms	remaining: 1m 13s
11:	total: 595ms	remaining: 1m 13s
12:	total: 649ms	remaining: 1m 14s
13:	total: 704ms	remaining: 1m 14s
14:	total: 755ms	remaining: 1m 14s
15:	total: 809ms	remaining: 1m 15s
16:	total: 860ms	remaining: 1m 14s
17:	total: 907ms	remaining: 1m 14s
18:	total: 964ms	remaining: 1m 15s
19:	total: 1s	remaining: 1m 14s
20:	total: 1.05s	remaining: 1m 14s
21:	total: 1.1s	remaining: 1m 13s
22:	total: 1.15s	remaining: 1m 13s
23:	total: 1.2s	remaining: 1m 13s
24:	total: 1.25s	remaining: 1m 13s
25:	total: 1.3s	remaining: 1m 13s
26:	total: 1.34s	remaining: 1m 13s
27:	total: 1.38s	remaining: 1m 12s
28:	total: 1.43s	remaining: 1m 12s

Default metric period is 5 because AUC is/are not implemented for GPU


4:	total: 253ms	remaining: 1m 15s
5:	total: 302ms	remaining: 1m 15s
6:	total: 353ms	remaining: 1m 15s
7:	total: 403ms	remaining: 1m 15s
8:	total: 454ms	remaining: 1m 15s
9:	total: 505ms	remaining: 1m 15s
10:	total: 561ms	remaining: 1m 15s
11:	total: 610ms	remaining: 1m 15s
12:	total: 662ms	remaining: 1m 15s
13:	total: 711ms	remaining: 1m 15s
14:	total: 766ms	remaining: 1m 15s
15:	total: 813ms	remaining: 1m 15s
16:	total: 861ms	remaining: 1m 15s
17:	total: 916ms	remaining: 1m 15s
18:	total: 962ms	remaining: 1m 14s
19:	total: 1.01s	remaining: 1m 14s
20:	total: 1.05s	remaining: 1m 14s
21:	total: 1.1s	remaining: 1m 13s
22:	total: 1.14s	remaining: 1m 13s
23:	total: 1.19s	remaining: 1m 13s
24:	total: 1.24s	remaining: 1m 13s
25:	total: 1.29s	remaining: 1m 13s
26:	total: 1.33s	remaining: 1m 12s
27:	total: 1.38s	remaining: 1m 12s
28:	total: 1.43s	remaining: 1m 12s
29:	total: 1.47s	remaining: 1m 12s
30:	total: 1.53s	remaining: 1m 12s
31:	total: 1.58s	remaining: 1m 12s
32:	total: 1.62s	remaining:

Default metric period is 5 because AUC is/are not implemented for GPU


0:	total: 54ms	remaining: 1m 20s
1:	total: 108ms	remaining: 1m 20s
2:	total: 160ms	remaining: 1m 19s
3:	total: 213ms	remaining: 1m 19s
4:	total: 264ms	remaining: 1m 19s
5:	total: 320ms	remaining: 1m 19s
6:	total: 366ms	remaining: 1m 18s
7:	total: 420ms	remaining: 1m 18s
8:	total: 468ms	remaining: 1m 17s
9:	total: 517ms	remaining: 1m 17s
10:	total: 643ms	remaining: 1m 27s
11:	total: 699ms	remaining: 1m 26s
12:	total: 760ms	remaining: 1m 26s
13:	total: 810ms	remaining: 1m 25s
14:	total: 858ms	remaining: 1m 24s
15:	total: 916ms	remaining: 1m 24s
16:	total: 972ms	remaining: 1m 24s
17:	total: 1.02s	remaining: 1m 24s
18:	total: 1.07s	remaining: 1m 23s
19:	total: 1.12s	remaining: 1m 22s
20:	total: 1.17s	remaining: 1m 22s
21:	total: 1.23s	remaining: 1m 22s
22:	total: 1.28s	remaining: 1m 22s
23:	total: 1.33s	remaining: 1m 22s
24:	total: 1.39s	remaining: 1m 21s
25:	total: 1.44s	remaining: 1m 21s
26:	total: 1.49s	remaining: 1m 21s
27:	total: 1.54s	remaining: 1m 21s
28:	total: 1.6s	remaining: 1m 2

Default metric period is 5 because AUC is/are not implemented for GPU


3:	total: 223ms	remaining: 1m 23s
4:	total: 282ms	remaining: 1m 24s
5:	total: 336ms	remaining: 1m 23s
6:	total: 390ms	remaining: 1m 23s
7:	total: 445ms	remaining: 1m 22s
8:	total: 498ms	remaining: 1m 22s
9:	total: 550ms	remaining: 1m 21s
10:	total: 596ms	remaining: 1m 20s
11:	total: 648ms	remaining: 1m 20s
12:	total: 700ms	remaining: 1m 20s
13:	total: 751ms	remaining: 1m 19s
14:	total: 800ms	remaining: 1m 19s
15:	total: 854ms	remaining: 1m 19s
16:	total: 909ms	remaining: 1m 19s
17:	total: 958ms	remaining: 1m 18s
18:	total: 1.01s	remaining: 1m 18s
19:	total: 1.06s	remaining: 1m 18s
20:	total: 1.11s	remaining: 1m 17s
21:	total: 1.16s	remaining: 1m 17s
22:	total: 1.21s	remaining: 1m 17s
23:	total: 1.26s	remaining: 1m 17s
24:	total: 1.31s	remaining: 1m 17s
25:	total: 1.36s	remaining: 1m 17s
26:	total: 1.41s	remaining: 1m 17s
27:	total: 1.46s	remaining: 1m 16s
28:	total: 1.52s	remaining: 1m 17s
29:	total: 1.57s	remaining: 1m 16s
30:	total: 1.62s	remaining: 1m 16s
31:	total: 1.67s	remaining: